Contribution form template improvements

gagnieray opened the pull request:
fixes #1960

trasher commented:
It works well with membership extension as far as I can see.
With a beginning of membership date, it does not (but proposed fix does not seem to affect that at all -- and I'm not even sure how this should work :() That therefore seems another issue.

Regarding UI changes, could you resume what have changed please?

trasher commented:
Seems not correct on contributions list. On current develop branch:
<img width="1560" height="412" alt="image" src="https://github.com/user-attachments/assets/b10cdac6-59d4-4f9f-8767-328eb7b2c532" />

With your changes:
<img width="1560" height="412" alt="image" src="https://github.com/user-attachments/assets/e651459d-e00e-4221-a37c-e0854f8d5438" />

I'm not against improvements on that part, but please do it in a separate PR we can merge after 1.2.0 release :)

gagnieray committed:
Contribution form template improvements

Update membership extension when contribution type is changed on new membership fee
creation.

Also add a glow transition effect on updated fields.

fixes #1960

gagnieray commented:
> I'm not against improvements on that part, but please do it in a separate PR we can merge after 1.2.0 release :)

Done !

FYI, on the actual PR, I added a glow transition effect on the fields when their values are updated on a contribution type change (amount and extension).

gagnieray commented:
> With a beginning of membership date, it does not (but proposed fix does not seem to affect that at all -- and I'm not even sure how this should work :() That therefore seems another issue.

Do you mean that with a beginning of membership date, the "end date of membership" on the form should also be update according to the duration defined in the settings of a contribution type (for exemple 4 months) ?

Then the other improvements for which I opened a separate PR are totally irrelevant, and that's indeed another issue.

trasher commented:
> > I'm not against improvements on that part, but please do it in a separate PR we can merge after 1.2.0 release :)
> 
> Done !

:+1: thank you!

> 
> FYI, on the actual PR, I added a glow transition effect on the fields when their values are updated on a contribution type change (amount and extension).

OK, no problem. I'll review this one quickly.

> > With a beginning of membership date, it does not (but proposed fix does not seem to affect that at all -- and I'm not even sure how this should work :() That therefore seems another issue.
> 
> Do you mean that with a beginning of membership date, the "end date of membership" on the form should also be update according to the duration defined in the settings of a contribution type (for exemple 4 months) ?

I guess it should, yes... It can cause issues with "offered months", but that's not been designed to work with a monthly membership.

Currently, if you first choose a monthly contribution, and then you change the selected member, the end date is changed - and that seem the correct way to go.
I'll open another issue on the tracker; no one yet complains about that problem. I even do not know if associations which uses a begin date of membership really uses monthly contributions types.

gagnieray commented:
> > Do you mean that with a beginning of membership date, the "end date of membership" on the form should also be update according to the duration defined in the settings of a contribution type (for exemple 4 months) ?
> 
> I guess it should, yes... It can cause issues with "offered months", but that's not been designed to work with a monthly membership.
> 
> Currently, if you first choose a monthly contribution, and then you change the selected member, the end date is changed - and that seem the correct way to go.

OK ! That makes sense :+1: 

> I'll open another issue on the tracker; no one yet complains about that problem. 

Fine !

> I even do not know if associations which uses a begin date of membership really uses monthly contributions types.

Indeed. Personnaly, I wrongly presumed that using a beginning of membership date always concerned yearly subscription. And that was the intent for the other PR #700 : to hide monthly choices in that configuration :sweat_smile:

trasher commented:
> Indeed. Personnaly, I wrongly presumed that using a beginning of membership date always concerned yearly subscription. And that was the intent for the other PR #700 : to hide monthly choices in that configuration 😅

Well, maybe you're right; I can't say :/
I guess we should ask on users mailing list to have feedback from concerned users?

gagnieray commented:
> I guess we should ask on users mailing list to have feedback from concerned users?

I guess yes. But what if users that use a monthly configuration, even if there is only one, didn't subscribed to the mailing list, and ask for a revert afterwards :confused: ?

I'll get a deeper look at this point :wink:
